extension/src/plots/model/util.ts

```diff
@@ -1,17 +1,24 @@
 import { getDataFromColumnPath } from '../../experiments/model/util'
 import { Experiment } from '../../experiments/webview/contract'
+import { RevisionFirstThreeColumns } from '../webview/contract'
 
 export const getRevisionFirstThreeColumns = (
   firstThreeColumns: string[],
   experiment: Experiment
 ) => {
-  const columns: Array<{ path: string; value: string }> = []
+  const columns: RevisionFirstThreeColumns = []
   for (const path of firstThreeColumns) {
-    const { value, splitUpPath } = getDataFromColumnPath(experiment, path)
-    const [type] = splitUpPath
+    const { value, splitUpPath, fullValue } = getDataFromColumnPath(
+      experiment,
+      path
+    )
+    const type = splitUpPath[0]
+
     if (value) {
       columns.push({
+        fullValue,
         path: path.slice(type.length + 1) || path,
+        type,
```

Comment:
[Q] Why not send only the raw data from the model and have the client(s) format it? See `webview/src/experiments/util/buildDynamicColumns.tsx` for how this is handled by the experiments table.

Reply:
If not there is some commonality with `ExperimentsTree.getTooltipTable`. The duplication should be removed/combined.

Reply:
It definitely would simplify the code a lot! Will do it in a followup :)